tests/test_mesh.py

```diff
@@ -0,0 +1,501 @@
+# -*- coding: utf-8 -*-
```

Comment:
This file has been moved and added to. The most relevant parts are starting at line 215.